app/src/components/widgets/multiLevelWidgets/mostPopularPatterns/mostPopularPatterns.jsx

```diff
@@ -58,27 +69,38 @@ export class MostPopularPatterns extends Component {
     }
   }
 
-  onBackClick = () => {
-    this.props.clearQueryParams(() => {
-      this.setState({
-        selectedPattern: null,
-      });
-    });
-  };
+  onPatternClick = (patternName) => {
+    const { widget, getPatternLink } = this.props;
+    const { selectedAttribute } = this.state;
 
-  onPatternClick = (pattern) => {
-    this.props.fetchWidget({ [PATTERN_FILTER_PARAM]: pattern }).then(() => {
-      this.setState({
-        selectedPattern: pattern,
-      });
+    const launchesLimit = widget.contentParameters.itemsCount;
+    const compositeAttribute = `${
+      widget.contentParameters.widgetOptions.attributeKey
+    }:${selectedAttribute}`;
+    const link = getPatternLink({
+      launchesLimit,
+      patternName,
+      compositeAttribute,
     });
+    const navigationParams = this.getDefaultParamsWidget(widget.appliedFilters[0].id);
```

Comment:
`getDefaultParamsWidget` sounds unclear. Please rename it to `getDefaultLinkParams` or `getCommonLinkParams`.